search-service/src/main/kotlin/com/egm/stellio/search/service/EntityTypeService.kt

```diff
@@ -0,0 +1,103 @@
+package com.egm.stellio.search.service
+
+import arrow.core.Either
+import arrow.core.flatten
+import arrow.core.left
+import arrow.core.right
+import com.egm.stellio.search.model.*
+import com.egm.stellio.search.util.*
+import com.egm.stellio.shared.model.APIException
+import com.egm.stellio.shared.model.ExpandedTerm
+import com.egm.stellio.shared.model.ResourceNotFoundException
+import com.egm.stellio.shared.util.AuthContextModel
+import com.egm.stellio.shared.util.JsonLdUtils.compactTerm
+import com.egm.stellio.shared.util.JsonLdUtils.compactTerms
+import com.egm.stellio.shared.util.JsonLdUtils.expandJsonLdTerm
+import com.egm.stellio.shared.util.typeNotFoundMessage
+import org.springframework.r2dbc.core.DatabaseClient
+import org.springframework.stereotype.Component
+
+@Component
+class EntityTypeService(
+    private val databaseClient: DatabaseClient
+) {
+
+    suspend fun getEntityTypeList(contexts: List<String>): EntityTypeList {
+        val entityTypes = databaseClient.sql(
+            """
+            SELECT DISTINCT(types) FROM entity_payload
+            ORDER BY types
+            """.trimIndent()
+        ).allToMappedList { rowToTypes(it) }.flatten()
```

Comment:
I'm wondering if you could have duplicates in your result.

For instance, if:
- Entity E1 has type Type T1 and Type T2
- Entity E2 has type Type T1

The distinct will return (T1, T2) and (T1), then you'll have T1, T2 and T1 in the final result?

Reply:
The solution will to transform array to string or add toSet at the end

Reply:
did you add a test for this use case?

Reply:
When we create entity payload the first contains BEEHIVE_TYPE and SENSOR_TYPE, the second only SENSOR_TYPE, and the last APIARY_TYPE. When we do a get to retrieve type list I receive only 3 types. It's ok this test ?

Reply:
LGTM